src/figuresClasses.ts

```diff
@@ -1,19 +1,111 @@
+enum Shape {
+  triangle = 'triangle',
+  circle = 'circle',
+  rectangle = 'rectangle',
+}
+
+enum Color{
+  red = 'red',
+  green = 'green',
+  blue = 'blue',
+}
+
+const INVALID_SIZE_ERROR = 'Invalid sides for this figure';
+
+function roundNumberToHundreds(num: number): number {
+  return Math.floor(num * 100) / 100;
+}
+
 export interface Figure {
+  shape: Shape,
+  color: Color,
 
+  getArea(): number
 }
 
-export class Triangle {
+export class Triangle implements Figure {
+  shape = Shape.triangle;
+
+  constructor(
+    public color: Color,
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    if (this.a <= 0 || this.b <= 0 || this.c <= 0) {
+      throw new Error(INVALID_SIZE_ERROR);
+    }
 
+    const sides = [a, b, c];
+    const biggestSide = Math.max(...sides);
+    const sumOfOtherSides = sides.reduce((acc, value) => {
+      if (value !== biggestSide) {
+        return acc + value;
+      }
+
+      return acc;
+    });
```

Comment:
No need to fix it - but a generic solution is not always the best one
IMO the best way to implement it is the simplest
```suggestion
    const isOk = a + b < c && a + c < b && b + c < a; 
```